osquery/main/shell.cpp

```diff
@@ -83,6 +89,43 @@ int profile(int argc, char *argv[]) {
   return 0;
 }
 
+// readline completion expects strings to be malloced. readline will free them
+// later.
+char *copy_string(const std::string &str) {
+  char *copy = (char *)malloc(str.size() + 1);
+  if (copy == nullptr) {
+    LOG(ERROR)
```

Comment:
Yeah, at this point it's definitely best effort. Writing directly to stderr might make it more likely to succeed?